provider/azure/environ.go

```diff
@@ -402,6 +405,25 @@ func (env *azureEnviron) ConstraintsValidator(ctx context.ProviderCallContext) (
 	return validator, nil
 }
 
+func getSliceOfAliasAndInstanceTypes(instanceTypes map[string]instances.InstanceType) ([]instances.InstanceType, []string) {
```

Comment:
@manadart Azure has constraint aliases. So that `standard_D2` can also be called `D2`.
So I had two choices:
- revert the change and just let it be
- split the map into its components (instances Slice, alias slice) and add them together
  - advantage: uses the common sorting logic
  - disadvantage: more code and logic added 

What step would you prefer?

Reply:
OK, let's leave this one as it was. We're getting fiddly without gaining much.